research/object_detection/dataset_tools/create_ava_actions_tf_record.py

```diff
@@ -0,0 +1,378 @@
+              if len(row) > 2 and int(row["action_label"]) in label_map:
+                boxes.append([float(row["ymin"]), float(row["xmin"]),
+                              float(row["ymax"]), float(row["xmax"])])
+                labels.append(int(row["action_label"]))
+                label_strings.append(label_map[int(row["action_label"])])
+                confidences.append(1)
+              else:
+                logging.warning("Unknown label: %s", row["action_label"])
+
+            total_boxes.append(boxes)
+            total_labels.append(labels)
+            total_label_strings.append(label_strings)
+            total_confidences.append(confidences)
+            windowed_timestamp += 1
+
+          if len(total_boxes) > 0:
+            yield seq_example_util.make_sequence_example("AVA", media_id, total_images,
+                int(height), int(width), 'jpeg', total_source_ids, None, total_is_annotated,
+                total_boxes, total_label_strings, use_strs_for_source_id=True)
+
+          #Move middle_time_frame, skipping excluded frames
+          frames_mv = 0
+          frames_excluded_count = 0
+          while (frames_mv < hop_between_sequences + frames_excluded_count
+                 and middle_frame_time + frames_mv < POSSIBLE_TIMESTAMPS[-1]):
+            frames_mv += 1
+            if (media_id, windowed_timestamp + frames_mv) in frame_excluded:
+              frames_excluded_count += 1
+          middle_frame_time += frames_mv
+
+        cur_vid.release()
+
+  def _download_data(self):
+    """Downloads and extracts data if not already available."""
+    if sys.version_info >= (3, 0):
+      urlretrieve = urllib.request.urlretrieve
+    else:
+      urlretrieve = urllib.request.urlretrieve
+    logging.info("Creating data directory.")
+    tf.io.gfile.makedirs(self.path_to_data_download)
+    logging.info("Downloading annotations.")
+    paths = {}
+    
+    zip_path = os.path.join(self.path_to_data_download,
+                            ANNOTATION_URL.split("/")[-1])
+    urlretrieve(ANNOTATION_URL, zip_path)
+    with zipfile.ZipFile(zip_path, 'r') as zip_ref:
+        zip_ref.extractall(self.path_to_data_download)
+    for split in ["train", "test", "val"]:
+      csv_path = os.path.join(self.path_to_data_download,
+                              "ava_%s_v2.2.csv" % split)
+      excl_name = "ava_%s_excluded_timestamps_v2.2.csv" % split
+      excluded_csv_path = os.path.join(self.path_to_data_download, excl_name)
+      SPLITS[split]["csv"] = csv_path
+      SPLITS[split]["excluded-csv"] = excluded_csv_path
+      paths[split] = (csv_path, excluded_csv_path)
+
+    label_map = self.get_label_map(os.path.join(self.path_to_data_download,
+                                                "ava_action_list_v2.2.pbtxt"))
+    return paths, label_map
+
+  def get_label_map(self, path):
+    """Parsess a label map into {integer:string} format."""
+    label_map = {}
+    with open(path, "r") as f:
```

Comment:
please use proto apis to parse label map. 

label_map_proto = StringIntLabelMap()
label_map_proro.FromString(label_map_content_as_str)

Reply:
I remember trying to use label_map_util but I think the ava label maps come in a different format so it can't be used.